students/allen_maxwell/Lesson_09/donor_models.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+# Allen Maxwell
+# Python 210
+# 1/20/2020
+# donor_models.py
+
+
+import os
+
+
+def check_name(name):
+    '''Checks the name entry for a Full Name (First and Last).'''
+    try:
+        name.split()[1]
+        name = name.title()
+        return name.title()
+    except TypeError:
+        raise ('Name must contain a first and last name')
+
+
+def check_donation(donation):
+    '''Checks the donation data for two positive values, > 0 [float, int].'''
+    if (donation[0] > 0 and donation[1] > 0):
+        donation[1] = int(donation[1])
+        return donation
+    else:
+        raise AttributeError('Donations must be positive values, > 0 [float, int]')
+
+
+class Donor(object):
+
+    def __init__(self, donor_name, donations):
+        '''Donor class input: Full Name, [total_donations, num_donations].'''
+        self._name = check_name(donor_name)
+        self._donations = check_donation(donations)
+
+    @property
+    def name(self):
+        '''Returns donor name'''
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        '''Blocks setting name of the donor.'''
+        raise AttributeError('Name cannot be set')
+
+    @property
+    def donations(self):
+        '''Returns the donation data.'''
+        return self._donations
+
+    @donations.setter
+    def donations(self, donation):
+        '''Blocks setting donations for the donor.'''
+        raise AttributeError('Donations cannot be set')
+
+    @property
+    def total_donations(self):
+        '''Returns total donation.'''
+        return self._donations[0]
+
+    @staticmethod
+    def add_total_donations(self, donation):
+        '''Adds the new donation value to the current total donations.'''
+        self._donations[0] = self.total_donations + self.check_donation_value(donation)
+
+    @staticmethod
+    def check_donation_value(donation):
+        '''Checks if the donation is greater than zero.'''
+        if donation > 0:
+            return donation
+        else:
+            raise AttributeError('Donations must be positive values, greater than zero')
+
+    @property
+    def num_donations(self):
+        '''Returns number donations.'''
+        return self._donations[1]
+
+    @staticmethod
```

Comment:
this wouldn't be a static method if you're using an instance via `self`

Reply:
Updated changes in Lesson 9 Assignment Update